app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/agent/BusinessListCYAController.scala

```diff
@@ -30,41 +30,53 @@ import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.forms.agent.AddAnot
 import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.services.AuthService
 import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.utilities.ImplicitDateFormatterImpl
-import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.views.html.agent.check_your_answers
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.views.html.agent.CheckYourAnswers
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
 
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.ExecutionContext
 
 @Singleton
 class BusinessListCYAController @Inject()(authService: AuthService,
+                                          checkYourAnswers: CheckYourAnswers,
                                           val incomeTaxSubscriptionConnector: IncomeTaxSubscriptionConnector,
                                           mcc: MessagesControllerComponents)
-                                         (implicit val ec: ExecutionContext,
-                                          val appConfig: AppConfig,
+                                         (implicit val appConfig: AppConfig,
+                                          val ec: ExecutionContext,
                                           dateFormatter: ImplicitDateFormatterImpl)
   extends FrontendController(mcc) with I18nSupport with ReferenceRetrieval {
 
   def view(addAnotherBusinessForm: Form[AddAnotherBusinessModel], businesses: Seq[SelfEmploymentData])(implicit request: Request[AnyContent]): Html = {
-    check_your_answers(
-      addAnotherBusinessForm = addAnotherBusinessForm,
-      answers = businesses,
-      postAction = routes.BusinessListCYAController.submit(),
-      implicitDateFormatter = dateFormatter
+    val appendable = checkYourAnswers(
+      addAnotherBusinessForm,
+      businesses,
+      routes.BusinessListCYAController.submit(),
+      dateFormatter
     )
+    appendable
   }
 
+  private val businesses1: Int = appConfig.limitOnNumberOfBusinesses
+
   def show: Action[AnyContent] = Action.async { implicit request =>
     authService.authorised() {
       withReference { reference =>
-        incomeTaxSubscriptionConnector.getSubscriptionDetails[Seq[SelfEmploymentData]](reference, businessesKey).map {
+        val future = incomeTaxSubscriptionConnector.getSubscriptionDetails[Seq[SelfEmploymentData]](reference, businessesKey)
+        future.map {
           case Right(Some(businesses)) if businesses.exists(_.isComplete) =>
-            Ok(view(addAnotherBusinessForm(businesses.size, appConfig.limitOnNumberOfBusinesses), businesses.filter(_.isComplete)))
-          case Right(_) => Redirect(routes.InitialiseController.initialise())
+            val size = businesses.size
+            val value = addAnotherBusinessForm(size, businesses1)
+            val datas = businesses.filter(_.isComplete)
+            val html = view(value, datas)
+            Ok(html)
```

Comment:
It might be good to simplify line 67 to 71:
```
val form = addAnotherBusinessForm(businesses.size, limitOnNumberOfBusinesses)
Ok(view(form, businesses.filter(_.isComplete)))
```